### PR TITLE
fixes #393: handling of boundary conditions for md-flexible

### LINKED ISSUE #393: VerletClusterCells::rebuild() creates segfault when particles are too far outside the domain
**Describe the bug**
In VerletClusterCells::rebuild() when particles are too far outside the domain, an invalid cell index is calculated resulting in a segmentation fault.

**To Reproduce**
Execute md-flexible with VerletClusterCells as only container option.
`./md-flexible --container VerletClusterCells`

### PR DESCRIPTION
**Pull request by SteffenSeckler**
# Description

- enables periodic boundary conditions by default for MDFlexConfig.
- fixes: calculate position update first, then boundary update.
- enables an error message if no boundary condition is set.
- ContainerSelector: changing a container now uses the owned state to specify whether to add a particle as owned or halo particle instead of using try-catch exception workflow.
- introduce isAppropriateClusterSize() and use them in VerletClusterCells.

## Resolved Issues

- [x] fixes #393 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] Adds an additional test for md-flex to checkExamples for VerletClusterCells.
- [x] Adds new unit test in IteratorTest.cpp that compares the iterators with and without openmp for all containers.

### CODE REVIEW

File: src/autopas/containers/verletListsCellBased/verletLists/VerletListHelpers.h [hunk 1]
```diff
@@ -251,6 +255,11 @@ class VerletListHelpers {
       return true;
     }
 
+    bool isAppropriateClusterSize(unsigned int clusterSize,
+                                  autopas::DataLayoutOption::Value dataLayout) const override {
```
Comment (FG-TUM) [inline src/autopas/containers/verletListsCellBased/verletLists/VerletListHelpers.h]:
```suggestion
                                  DataLayoutOption::Value dataLayout) const override {
```

Comment (FG-TUM) [reply src/autopas/containers/verletListsCellBased/verletLists/VerletListHelpers.h]: will need reformat after applying this

File: tests/testAutopas/tests/autopasInterface/Newton3OnOffTest.cpp [hunk 2]
```diff
@@ -32,7 +32,7 @@ INSTANTIATE_TEST_SUITE_P(
 
               // container factory
               autopas::ContainerSelector<Particle, FPCell> containerSelector({0., 0., 0.}, {10., 10., 10.}, 1.);
-              autopas::ContainerSelectorInfo containerInfo(1., 0.);
+              autopas::ContainerSelectorInfo containerInfo(1., 0., 64);
```
Comment (FG-TUM) [inline tests/testAutopas/tests/autopasInterface/Newton3OnOffTest.cpp]: is 64 really what we want here?

Comment (SteffenSeckler) [reply tests/testAutopas/tests/autopasInterface/Newton3OnOffTest.cpp]: it's sth. that works with cuda, so yes !

Comment (SteffenSeckler) [reply tests/testAutopas/tests/autopasInterface/Newton3OnOffTest.cpp]: and it was the previous default value.

**Pull request merged**